evap/staff/tests/test_views.py

```diff
@@ -1682,3 +1682,43 @@ def test_degree_form(self):
         self.assertIn("Successfully", str(response))
 
         self.assertTrue(Degree.objects.filter(name_de="Test", name_en="Test").exists())
+
+
+class TestSemesterQuestionnaireAssignment(ViewTest):
+    @classmethod
+    def setUpTestData(cls):
+        mommy.make(UserProfile, username='staff', groups=[Group.objects.get(name='Staff')])
+        cls.semester = mommy.make(Semester)
+        cls.course_type_1 = mommy.make(CourseType, name_en="ct1")
+        cls.course_type_2 = mommy.make(CourseType, name_en="ct2")
+        cls.responsible = mommy.make(UserProfile)
+        cls.questionnaire_1 = mommy.make(Questionnaire, type=Questionnaire.TOP, name_en="q1")
+        cls.questionnaire_2 = mommy.make(Questionnaire, type=Questionnaire.TOP, name_en="q2")
+        cls.questionnaire_responsible = mommy.make(Questionnaire, type=Questionnaire.CONTRIBUTOR, name_en="qr")
+        cls.course_1 = mommy.make(Course, semester=cls.semester, type=cls.course_type_1)
+        cls.course_2 = mommy.make(Course, semester=cls.semester, type=cls.course_type_2)
+        mommy.make(Contribution, contributor=cls.responsible, course=cls.course_1, responsible=True, can_edit=True, comment_visibility=Contribution.ALL_COMMENTS)
+        mommy.make(Contribution, contributor=cls.responsible, course=cls.course_2, responsible=True, can_edit=True, comment_visibility=Contribution.ALL_COMMENTS)
+
+    def test_questionnaire_assignment(self):
+        page = self.get_assert_200("/staff/semester/{}/assign".format(self.semester.id), user="staff")
+        form = page.forms["questionnaire-assign-form"]
+        form[self.course_type_1.name] = [self.questionnaire_1.pk, self.questionnaire_2.pk]
+        form[self.course_type_2.name] = [self.questionnaire_2.pk]
+        form["Responsible contributor"] = [self.questionnaire_responsible.pk]
+
+        response = form.submit()
+        self.assertIn("Successfully", str(response))
+
+        self.assertEqual(self.course_1.general_contribution.questionnaires.count(), 2)
+        self.assertTrue(self.course_1.general_contribution.questionnaires.filter(id=self.questionnaire_1.id).exists())
+        self.assertTrue(self.course_1.general_contribution.questionnaires.filter(id=self.questionnaire_2.id).exists())
```

Comment:
`self.assertEqual(set(self.course_1.general_contribution.questionnaires), set([self.questionnaire_1, self.questionnaire_2])`?

similar for the others, either use lists with one element there or `.get` on the queryset and compare with the questionnaire directly.